src/Parser.php

```diff
@@ -38,90 +40,74 @@ class Parser
     /**
      * @var Table[]
      */
-    private $csvFiles = [];
+    private array $csvFiles = [];
 
-    /**
-     * @var Cache
-     */
-    private $cache;
+    private Cache $cache;
 
-    /**
-     * @var Temp
-     */
-    private $temp;
+    private Temp $temp;
 
-    /**
-     * @var array
-     */
-    private $primaryKeys = [];
+    private array $primaryKeys = [];
 
-    /**
-     * @var Analyzer
-     */
-    private $analyzer;
+    private Analyzer $analyzer;
 
-    /**
-     * @var Structure
-     */
-    private $structure;
+    private Structure $structure;
 
     /**
-     * Parser constructor.
-     * @param Analyzer $analyzer
      * @param array $definitions should contain an array with previously
      *         cached results from analyze() calls (called automatically by process())
      */
     public function __construct(Analyzer $analyzer, array $definitions = [])
     {
-        ini_set('serialize_precision', 17);
+        ini_set('serialize_precision', '17');
         $this->analyzer = $analyzer;
         $analyzer->getStructure()->load($definitions);
         $this->structure = $analyzer->getStructure();
-        $this->temp = new Temp("json-parser");
+        $this->temp = new Temp('json-parser');
         $this->cache = new Cache();
     }
 
+    public function __destruct()
+    {
+        $this->temp->remove();
```

Comment:
Tu stoji za zmienku, ze temp v1 mazal automaticky adresare, ale v temp v2 to treba zmazat rucne.